### PR TITLE
Add support for output directory (psardecrypt only!)

### PR DESCRIPTION
**Pull request by zecoxao**
This adds support for the output directory option (requiring a second paramenter to specify where you want to decrypt your files)

### CODE REVIEW

**Comment by artart78** [comment]
I would remove the F0 subdirectory as it's the only subdirectory of the output directory.

File: ipl_decrypt.cpp [hunk 1]
```diff
@@ -550,18 +551,18 @@ int decryptIPL(u8 *inData, u32 inDataSize, int version, u32 loadAddr, const char
                 if (debug) {
                     printf("decompressed %d bytes\n", decSize);
                 }
-                sprintf(szDataPath, "./F0/PSARDUMPER/stage2_%s.gz", filename);
```
Comment (John-K) [inline ipl_decrypt.cpp:553]: `szDataPath` is likely not large enough any more. Should also be using `snprintf` to be sure that we always null terminate as well.

File: psardecrypt.cpp [hunk 2]
```diff
@@ -12,11 +12,12 @@ int
 main(int argc, char *argv[]) {
 	int outSize;
 
-	if (argc != 2) {
-		printf("Usage: %s <infile>\n", basename(argv[0]));
+	if (argc != 3) {
+		printf("Usage: %s <infile> <outdir> \n", basename(argv[0]));
```
Comment (John-K) [inline psardecrypt.cpp:16]: We should make this optional instead of mandatory

**Review by John-K — COMMENTED** [review]
@zecoxao would you mind taking a look at some of the comments and resolving them if you can?

File: ipl_decrypt.cpp (hunk 1 above)
Comment (John-K) [reply ipl_decrypt.cpp:553]: Also consider making `szDataPath` a `std::string`

File: psardecrypt.cpp (hunk 2 above)
Comment (zecoxao) [reply psardecrypt.cpp:16]: No idea on how to make it optional tho. i assume it'd require a lot of changes in order to make it work

Comment (zecoxao) [reply psardecrypt.cpp]: Ok, i have used a clever trick to make it optional, it'll work this way i think.

File: ipl_decrypt.cpp (hunk 1 above)
Comment (zecoxao) [reply ipl_decrypt.cpp:553]: i have fixed that in the second to last commit :)

Comment (artart78) [reply ipl_decrypt.cpp:553]: The issue remains at some places though.
I'd suggest changing szDataPath to a `std::string` as John-K suggested, then replace
`strcpy(szDataPath, (outdir + "/F0/PSARDUMPER/12000_files_table.bin").c_str());`
with `dataPath = outdir + "/F0/PSARDUMPER/12000_files_table.bin"`
and
`sprintf(szDataPath, (outdir + "/F0/PSARDUMPER/stage3_%s").c_str(), filename);`
with `dataPath = outdir + "/F0/PSARDUMPER/stage3_" + std::string(filename)` (hoping that works as I hope it does)

Comment (zecoxao) [reply ipl_decrypt.cpp:553]: i think it should be fixed on the other commits i've made just now

File: PsarDecrypter.cpp [hunk 3]
```diff
@@ -558,19 +558,20 @@ int pspDecryptPSAR(u8 *dataPSAR, u32 size)
...
-    mkdir("./F0/font", 0777);
-    mkdir("./F0/kd", 0777);
-    mkdir("./F0/vsh", 0777);
-    mkdir("./F0/data/cert", 0777);
-    mkdir("./F0/kd/resource", 0777);
-    mkdir("./F0/vsh/etc", 0777);
-    mkdir("./F0/vsh/module", 0777);
-    mkdir("./F0/vsh/resource", 0777);
-    mkdir("./F0/codepage", 0777);
+	mkdir((outdir).c_str(), 0777);
```
Comment (John-K) [inline PsarDecrypter.cpp:561]:
```suggestion
    mkdir((outdir).c_str(), 0777);
```

**Review by John-K — APPROVED** [review]
Looks good, thanks for the edits!

**Pull request merged**